packages/rrweb/src/record/iframe-manager.ts

```diff
@@ -98,6 +111,15 @@ export class IframeManager {
     const iframeSourceWindow = message.source;
     if (!iframeSourceWindow) return;
 
+    if (
+      iframeSourceWindow == window.parent &&
+      window != window.parent &&
+      message.data.snapshot
```

Comment:
Annoyingly, typescript (eslint) is failing here; you could define your own message type (see CrossOriginIframeMessageEventContent) and cast to that message type with the `as` keyword, or maybe the following will work

```suggestion
            'snapshot' in message.data
```